packages/paste-website/src/pages/components/editable-code-block/index.mdx

```diff
@@ -0,0 +1,370 @@
+  title="Multi-file Editable Code Block"
+  storyID="components-editable-code-block--multi-file-editor"
+  code={`<Stack orientation="vertical" spacing="space40">
+  <RadioButtonGroup
+    attached
+    name="file-selector"
+    legend="Select file"
+    helpText="Show code for the selected file"
+    onChange={(_fileKey) => setFileKey(_fileKey)}
+  >
+    {Object.keys(Files).map((key) => (
+      <RadioButton key={key} value={key} defaultChecked={fileKey === key}>
+        {Files[key].name}
+      </RadioButton>
+    ))}
+  </RadioButtonGroup>
+  <EditableCodeBlockWrapper>
+    <EditableCodeBlockHeader>{Files[fileKey].name}</EditableCodeBlockHeader>
+    <EditableCodeBlock
+      height="400px"
+      path={Files[fileKey].name}
+      defaultLanguage={Files[fileKey].language}
+      defaultValue={Files[fileKey].value}
+    />
+  </EditableCodeBlockWrapper>
+</Stack>`}
+/>
+
+## States
+
+### Read only
+
+If the user doesn't have permission to edit the code, use the `readOnly` prop to make the Editable Code Block read-only.
+
+<StoryPreview
+  title="Read-Only Editable Code Block"
+  storyID="components-editable-code-block--read-only"
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>ReadOnly Example</EditableCodeBlockHeader>
+  <EditableCodeBlock height="45vh" readOnly defaultLanguage="typescript" defaultValue={TypeScriptExample} />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Syntax errors
+
+Monaco Editor will detect and show syntax errors out of the box.
+
+<StoryPreview
+  title="Syntax error example in Editable Code Block"
+  storyID="components-editable-code-block--syntax-error"
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>Syntax error example</EditableCodeBlockHeader>
+  <EditableCodeBlock
+    height="45vh"
+    defaultLanguage="typescript"
+    defaultValue={CodeStringWithSyntaxError}
+  />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Line-specific errors
+
+Show line-specific errors to help users pinpoint what line of code they need to fix.
+
+<StoryPreview
+  title="Custom line-specific errors in Editable Code Block"
+  storyID="components-editable-code-block--custom-inline-error"
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>Custom inline error example (try hovering it)</EditableCodeBlockHeader>
+  <EditableCodeBlock
+    height="45vh"
+    onChange={(value) => console.log(value)}
+    inlineErrorRange={{
+      startLineNumber: 3,
+      endLineNumber: 3,
+      startColumn: 7,
+      endColumn: 13,
+    }}
+    inlineErrorIsWholeLine={false}
+    inlineErrorHoverMessage={{value: '"id" can only be a string type.'}}
+    defaultLanguage="typescript"
+    defaultValue={TypeScriptExample}
+  />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Generic errors
+
+If there’s an error that can’t be traced to specific line numbers, use a [Callout](/components/callout).
+
+For additional guidance on how to compose error messages, refer to the [error state pattern](/patterns/error-state).
```

Comment:
```suggestion
For additional guidance on how to compose error messages, refer to the [Error State Pattern](/patterns/error-state).
```

Reply:
We keep patterns lowercase, generally, to help differentiate them from components which are like built Things

Reply:
news to me lol